### PR TITLE
[core] [bug] Partition should not be filtered when manifest file predicate and file itself all min/max is null

### PR DESCRIPTION
**Pull request by leaves12138**
### Purpose

Linked issue: close #xxx

### CODE REVIEW

**Review by JingsongLi — COMMENTED** [review]
See `PredicateBuilder.equalPartition` too.

**Comment by JingsongLi** [comment]
See `RowDataToObjectArrayConverter.createEqualPredicate`.

**Comment by JingsongLi** [comment]
See `SnapshotReaderImpl.withPartitionFilter`.

**Comment by leaves12138** [comment]
FileStoreLookupFunction.createSpecificPartFilter

**Comment by leaves12138** [comment]
PredicateBuilder.partition(Map<String, String> map, RowType rowType)

File: paimon-core/src/main/java/org/apache/paimon/partition/PartitionPredicate.java [hunk 1]
```diff
@@ -121,8 +121,11 @@ private MultiplePartitionPredicate(
             PredicateBuilder builder = new PredicateBuilder(partitionType);
             for (int i = 0; i < collectors.length; i++) {
                 SimpleColStats stats = collectors[i].result();
-                min[i] = builder.greaterOrEqual(i, stats.min());
-                max[i] = builder.lessOrEqual(i, stats.max());
+                Object minValue = stats.min();
+                Object maxValue = stats.max();
+
+                min[i] = minValue == null ? builder.isNull(i) : builder.greaterOrEqual(i, minValue);
+                max[i] = maxValue == null ? builder.isNull(i) : builder.lessOrEqual(i, maxValue);
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/partition/PartitionPredicate.java:128]: Seems wrong logic here.
Collector should not collect null values.

Comment (JingsongLi) [reply paimon-core/src/main/java/org/apache/paimon/partition/PartitionPredicate.java:128]: I see, if it is null, all values are null.

**Comment by JingsongLi** [comment]
+1

**Pull request merged**